### ISSUE TITLE
GDB Pretty Printing of Vector depends on deprecated member

### ISSUE THREAD

**Issue by TLJB** [dealii/dealii#12522]
I have recently tried to use the pretty printer functions for gdb supplied in contrib/utilities/dotgdbinit.py  to print the values
of a Vector object. This fails with the error message:
````
Python Exception <class 'gdb.error'> There is no member or method name data_end.:
````
This problem first ocured after upgrading from version 9.2 to  #9.3.
Other pretty printer functions (for example for tensors and points) function as expected.
After some digging I've found that gdb tries to use the members data_end and data_begin of the AlignedVector class 
to calculate the length of the vector and to iterate over the container.
Both members where present in deal 9.2 but seem to have been deprecated in version 9.3.

As such the pretty printer function no longer works.

In order to reproduce the problem install the pretty printer functions as described [here](https://www.dealii.org/9.2.0/users/gdb.html).
Then debug the step-3 tutorial using gdb and try to print the system_rhs vector.

**Comment by bangerth** [comment]
True. The same is the case for the `AlignedVector` class. We ought to fix this.

@TLJB Is this something you'd like to try your hand on? I think this could make for a good patch!

**Comment by TLJB** [comment]
I fixed the issue and made a pull request ( #12546 ).
This is my first time contributing so if anything is not up to code please let me know and I'll try to fix it.

**Comment by bangerth** [comment]
I double checked both the `Vector` pretty printer and all of the others, and they all currently work. So closing.
